source/extensions/tracers/opentelemetry/resource_detectors/dynatrace/dynatrace_resource_detector.cc

```diff
@@ -0,0 +1,57 @@
+#include "source/extensions/tracers/opentelemetry/resource_detectors/dynatrace/dynatrace_resource_detector.h"
+
+#include <fstream>
+#include <iostream>
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace OpenTelemetry {
+namespace {
+
+void addAttributes(const std::string& content, Resource& resource) {
+  for (const auto& line : StringUtil::splitToken(content, "\n", false, true)) {
+    const auto keyValue = StringUtil::splitToken(line, "=");
+    if (keyValue.size() != 2) {
+      continue;
+    }
+
+    const std::string key = std::string(keyValue[0]);
+    const std::string value = std::string(keyValue[1]);
+    resource.attributes_[key] = value;
+  }
+}
+
+} // namespace
+
+Resource DynatraceResourceDetector::detect() {
+  Resource resource;
+  resource.schemaUrl_ = "";
+  int failureCount = 0;
+
+  for (const auto& file_name : DT_METADATA_FILES) {
+    TRY_NEEDS_AUDIT {
```

Comment:
Can we just use StatusOr to propagate errors instead of exception handling in the data plane?

Reply:
As I mentioned below, this is called when Envoy is starting up, while creating the Tracer. From previous PRs, I learned that this is the "configuration path/ingest" https://github.com/envoyproxy/envoy/pull/29207#discussion_r1315348130